react/class_components/class_based_components.md

```diff
@@ -0,0 +1,301 @@
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- A note on the relevance of class components.
+- Covers the structure of a class component, the uses of the constructor and render method.
+- Discusses the ways to use props and state in class components.
+- Highlights the uses of `this` in class components.
+
+### The What's And Why's
+
+In your previous lessons, you have already been introduced to functional components, and the basic patterns in which components get written now a days. However, React components did not look this way since React was introduced.
+
+If you look into any older React codebase, you'll notice a lot of classes. These are known as class-based components. Prior to February 2019, functional components were also called state-less, as there was no way to manage state in them. This was changed when hooks were introduced, leading to less verbose and 'neater' components.
+
+In your career, chances are, you will be dealing with legacy code, so there will be days where you would be dealing with class components. To meet this issue, we take a peek into the works and intricacies of a class-based component, and how they work.
+
+### Building A Class Component
+
+As we already know about functional components, let us build a class-based component from a functional one. Usually, you will want to divide the contents of a component, like the one we use, into smaller, re-usable components, but for the purposes of this exercise, we stick to one component. Below, we have a sample functional component:
+
+~~~jsx
+import React, { useState } from "react";
+
+const FunctionalInput = ({ name }) => {
+  /*
+    We declare two state variables and their setters,
+    one to store the To-Do's
+    and the other to store the value of the input field
+  */
+  const [todos, setTodos] = useState(["Just some demo tasks", "As an example"]);
+  const [inputVal, setInputVal] = useState("");
+
+  const handleInputChange = (e) => {
+    setInputVal(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setTodos((todo) => [...todo, inputVal]);
+    setInputVal("");
+  };
+
+  return (
+    <section>
+      <h3>{name}</h3>
+      {/* The input field to enter To-Do's */}
+      <form onSubmit={handleSubmit}>
+        <label htmlFor="task-entry">Enter a task: </label>
+        <input
+          type="text"
+          name="task-entry"
+          value={inputVal}
+          onChange={handleInputChange}
+        />
+        <button type="submit">Submit</button>
+      </form>
+      <h4>All the tasks!</h4>
+      {/* The list of all the To-Do's, displayed */}
+      <ul>
+        {todos.map((todo) => (
+          <li key={todo}>{todo}</li>
+        ))}
+      </ul>
+    </section>
+  );
+};
+
+export default FunctionalInput;
+~~~
```

Comment:
We should remove comments in this code example. By this point, learners should be well versed in functional components and how state works.